src/github_runner_manager/openstack_cloud/health_checks.py

```diff
@@ -0,0 +1,241 @@
+#  Copyright 2024 Canonical Ltd.
```

Comment:
Linter complained about too many lines in `openstack_runner_manager`. So I extracted the health checks into a separate module.